### PR TITLE
MultiAgentDebate for predictions

### PR DESCRIPTION
**Pull request by namesty**

### CODE REVIEW

File: evo_prophet/functions/debate_prediction.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+{additional_information}
+
+I made the following prediction:
+
+```
+{prediction_0}
+```
+
+And you made ther following prediction:
```
Comment (kongzii) [inline evo_prophet/functions/debate_prediction.py:54]:
```suggestion
And you made the following prediction:
```

**Comment by kongzii** [comment]
Interesting!

Code LGTM, do you have any specific benchmark results or probably not yet?

**Review by kongzii — APPROVED** [review]

**Comment by evangriffiths** [comment]
Would also be interested to see the outcomes you get from this system. Especially because it's not super obvious to me that future even prediction (probabilistic answers) benefits in that same way as the kinds of problems they test on in that paper (maths/reasoning problems, which have a certain, knowable answer). If the limiting factor is in the prediction step, then hopefully we'll see an improvement here. But if the limiting factor is in the quality of the research, then maybe the performance won't be that different. Either way, very interested to see the results!

**Comment by namesty** [comment]
> Would also be interested to see the outcomes you get from this system. Especially because it's not super obvious to me that future even prediction (probabilistic answers) benefits in that same way as the kinds of problems they test on in that paper (maths/reasoning problems, which have a certain, knowable answer). If the limiting factor is in the prediction step, then hopefully we'll see an improvement here. But if the limiting factor is in the quality of the research, then maybe the performance won't be that different. Either way, very interested to see the results!

Yeah, I think these are great points. I'll merge it as a function; and now that it's working, let's benchmark it and see if it tangibly helps

**Pull request merged**